test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -1484,16 +1483,20 @@ private void testBlockLoader(boolean syntheticSource, boolean columnReader) thro
             );
         }
         var sourceLoader = mapper.mappingLookup().newSourceLoader(null, SourceFieldMetrics.NOOP);
-        testBlockLoader(columnReader, example, blockReaderSupport, sourceLoader);
+        testBlockLoader(FieldExtractPreference.forColumnReader(columnReader), example, blockReaderSupport, sourceLoader);
     }
 
     protected final void testBlockLoader(
-        boolean columnReader,
+        FieldExtractPreference fieldExtractPreference,
         SyntheticSourceExample example,
         BlockReaderSupport blockReaderSupport,
         SourceLoader sourceLoader
     ) throws IOException {
-        BlockLoader loader = blockReaderSupport.getBlockLoader(columnReader);
+        var columnReader = switch (fieldExtractPreference) {
```

Comment:
Not really. My comment here was about continuing to use fieldExtractPreference, which we agreed on.

Reply:
However, I don't understand why `EXTRACT_SPATIAL_BOUNDS` has columnReader=false? It should be true, because we'll be reading doc-values, which is column oriented.

Reply:
Well, it's mostly to avoid triggering the old behavior. Ideally nothing would use that Boolean, but since some old code *does* use it, I want this to take a different path.

Reply:
OK, upon re-reviewing the code, I see now what you mean. I've changed it to `EXTRACT_SPATIAL_BOUNDS` is column reader.

Reply:
Perhaps this code is simply never tested, which is why we see no errors here?

Reply:
Yes, nothing is exercising the path of `EXTRACT_SPATIAL_BOUNDS`. I think I was trying to use it before I gave up on it and just tested `AbstractShapeGeometryFieldMapper` directly! Let me revert this whole thing back to Boolean. No reason to have stamp coupling :)